web/src/compose_ui.ts

```diff
@@ -148,6 +149,51 @@ export function rewire_insert_and_scroll_into_view(
     insert_and_scroll_into_view = value;
 }
 
+export function maybe_show_scrolling_formatting_buttons(container_selector: string): void {
+    const button_container = document.querySelector(container_selector);
+    const button_bar = document.querySelector(
+        `${container_selector} .compose-control-buttons-container`,
+    );
+
+    if (!button_container || !button_bar) {
+        return;
+    }
+
+    const button_container_width = button_container?.clientWidth;
+    const button_bar_width = button_bar?.scrollWidth;
+    const button_bar_scroll_left = button_bar?.scrollLeft;
+
+    const button_bar_max_left_scroll = button_bar_width - button_container_width;
+
+    assert(
+        typeof button_container_width === "number" &&
+            typeof button_bar_width === "number" &&
+            typeof button_bar_scroll_left === "number",
+    );
+
+    // Set these values as data attributes for ready access by
+    // other scrolling logic
+    if (button_container instanceof HTMLElement) {
+        button_container.dataset.buttonContainerWidth = button_container_width.toString();
+        button_container.dataset.buttonBarWidth = button_bar_width.toString();
+        button_container.dataset.buttonBarMaxLeftScroll = button_bar_max_left_scroll.toString();
```

Comment:
I've been preferring to not use `dataset` over using `.attr("data-"` because `git grep` handles it better without the case format transition.

Reply:
That's a good point. I can refactor for that.

Reply:
We have a recommended eslint/unicorn rule that's forcing the use of `dataset` over `setAttribute`. I've opened [a CZO discussion](https://chat.zulip.org/#narrow/channel/6-frontend/topic/ESLint.20rule.3A.20unicorn.2Fprefer-dom-node-dataset/near/2039698) about this.